janusgraph-core/src/main/java/org/janusgraph/core/JanusGraphFactory.java

```diff
@@ -104,7 +116,60 @@ public static JanusGraph open(BasicConfiguration configuration) {
      * @return JanusGraph graph database
      */
     public static JanusGraph open(ReadConfiguration configuration) {
-        return new StandardJanusGraph(new GraphDatabaseConfiguration(configuration));
+        return open(configuration, null);
+    }
+
+    public static JanusGraph open(ReadConfiguration configuration, String backupName) {
+        final ModifiableConfiguration config = new ModifiableConfiguration(ROOT_NS, (WriteConfiguration) configuration, BasicConfiguration.Restriction.NONE);
+        final String graphName = config.has(GRAPH_NAME) ? config.get(GRAPH_NAME) : backupName;
+        if (null != graphName) {
+            JanusGraphManager jgm = JanusGraphManager.getInstance();
+            Preconditions.checkState(jgm != null, JANUS_GRAPH_MANAGER_EXPECTED_STATE_MSG);
+            return (JanusGraph) jgm.openGraph(graphName, (gName) -> {
+                return new StandardJanusGraph(new GraphDatabaseConfiguration(configuration));
+            });
+        } else {
+            log.info("You should supply \"graph.graphname\" in your .properties file configuration if you are opening " +
```

Comment:
Can you clarify? I'm talking about users that are accessing graphs directly, not remotely through GremlinServer. For example say they just have a graph that they're connecting to directly in a Gremlin shell or in their Java/Scala client app, with no Gremlin server deployed at all. In this case `jgm` would be expected to be null and it would be appropriate for `graph.graphname` to be unset, right? Or are you saying that accessing graphs directly without going through GremlinServer is an unusual access pattern and should be deprecated?